### PR TITLE
Save/load

### PR DESCRIPTION
**Pull request by DierF**
Fix bugs.
Add tests for serialize/deserialize.
Cleanup serializer.

### CODE REVIEW

File: src/umleditor/mvc_model/relation.py [hunk 1]
```diff
@@ -108,4 +108,4 @@ def __eq__(self, other):
         if self._destination != other._destination:
             return False
         
-        return True
\ No newline at end of file
+        return self._type == other._type
```
Comment (AdamG-L) [inline src/umleditor/mvc_model/relation.py:111]: Either don't type check or add helper method that replaces this check wherever this overload is being used

**Review by AdamG-L — CHANGES_REQUESTED** [review]
Either don't type check or add helper method that replaces this check wherever this overload is being used

File: src/umleditor/mvc_model/relation.py (hunk 1 above)
Comment (DierF) [reply src/umleditor/mvc_model/relation.py:125]: Fixed

**Review by AdamG-L — APPROVED** [review]
Approved, updated way of checking for equality minus types looks good

**Pull request merged**